src/main/java/com/autotune/analyzer/services/SearchSpace.java

```diff
@@ -37,91 +37,107 @@
+     *                 "lower_bound": "1",
+     *                 "name": "quarkus.thread-pool.core-threads",
+     *                 "step": 1,
+     *                 "upper_bound": "10"
+     *             },
+     *             {
+     *                 "value_type": "integer",
+     *                 "lower_bound": "1",
+     *                 "name": "quarkus.thread-pool.queue-size",
+     *                 "step": 1,
+     *                 "upper_bound": "100"
+     *             },
+     *             {
+     *                 "value_type": "integer",
+     *                 "lower_bound": "1",
+     *                 "name": "quarkus.hibernate-orm.jdbc.statement-fetch-size",
+     *                 "step": 1,
+     *                 "upper_bound": "50"
+     *             }
+     *         ],
+     *         "direction": "minimize"
+     *     }
      * ]
+     *
      * @param req
      * @param resp
      * @throws IOException
      */
     @Override
     protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
         JSONArray outputJsonArray = new JSONArray();
-        resp.setContentType("application/json");
+        resp.setContentType(JSON_CONTENT_TYPE);
 
-        String applicationName = req.getParameter(AnalyzerConstants.ServiceConstants.APPLICATION_NAME);
+        String experimentName = req.getParameter(AnalyzerConstants.ServiceConstants.EXPERIMENT_NAME);
+        String containerImageName = req.getParameter(AnalyzerConstants.ServiceConstants.STACK_NAME);
 
-        for (String autotuneObjectKey : AutotuneDeployment.applicationServiceStackMap.keySet()) {
-            AutotuneObject autotuneObject = AutotuneDeployment.autotuneObjectMap.get(autotuneObjectKey);
+        if (AutotuneDeployment.autotuneObjectMap.isEmpty()) {
+            outputJsonArray.put(AUTOTUNE_OBJECTS_NOT_FOUND);
+            resp.getWriter().println(outputJsonArray.toString(4));
+            return;
+        }
 
-            if (applicationName == null) {
-                //No application parameter, generate search space for all applications
-                for (String application : AutotuneDeployment.applicationServiceStackMap.get(autotuneObjectKey).keySet()) {
-                    addApplicationToSearchSpace(outputJsonArray, autotuneObjectKey, autotuneObject, application);
-                }
-            } else {
-                if (AutotuneDeployment.applicationServiceStackMap.get(autotuneObjectKey).containsKey(applicationName)) {
-                    addApplicationToSearchSpace(outputJsonArray, autotuneObjectKey, autotuneObject, applicationName);
+        if (!AutotuneDeployment.applicationServiceStackMap.isEmpty()) {
+            for (String autotuneObjectKey : AutotuneDeployment.applicationServiceStackMap.keySet()) {
+                AutotuneObject autotuneObject = AutotuneDeployment.autotuneObjectMap.get(autotuneObjectKey);
+
+                if (containerImageName == null) {
```

Comment:
Shouldn't this be experiment_name?

Reply:
hmmm good catch, looks like I messed up the searchSpace.java and copied over the wrong file, it is missing an entire section.